File: notes-server/models/note.go
package models

// GenericNote 通用笔记结构，字段由 schema 动态定义
type GenericNote struct {
	TypeID    string                 `json:"typeId"`
	ID        string                 `json:"id"`
	Fields    map[string]interface{} `json:"fields"`
	Content   string                 `json:"content,omitempty"`
	SubFiles  map[string]string      `json:"subFiles,omitempty"`
	Path      string                 `json:"path"`
	UpdatedAt string                 `json:"updatedAt,omitempty"`
}

File: notes-web/src/types/note.ts
/**
 * 通用 Note 类型（schema-driven）
 */

// 通用笔记（字段由 schema 动态定义）
export interface GenericNote {
  typeId: string
  id: string
  fields: Record<string, unknown>
  content?: string
  subFiles?: Record<string, string>
  path: string
  updatedAt?: string
}
